recipes/cpr/all/conanfile.py

```diff
@@ -67,21 +88,34 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+        ssl_library = self._get_ssl_library()
+        if not self._can_auto_ssl and ssl_library == CprConan._AUTO_SSL:
+            if self._supports_openssl:
+                self.output.info("Auto SSL is not available below version 1.6.0. Falling back to openssl")
+                self.options.with_ssl = "openssl"
+            else:
+                self.output.info("Auto SSL is not available below version 1.6.0 (or below 1.6.2 on macOS), and openssl not supported. Disabling SSL")
+                self.options.with_ssl = CprConan._NO_SSL
+
     def configure(self):
+        SSL_CURL_LIBS = {
+            "openssl": "openssl",
+            "darwinssl": "darwinssl",
+            "winssl": "schannel"
+        }
+
         if self.options.shared:
             del self.options.fPIC
         if not self._supports_openssl:
             del self.options.with_openssl
         if not self._supports_winssl:
             del self.options.with_winssl
 
+        ssl_library = self._get_ssl_library()
         # Make sure libcurl uses the same SSL implementation
-        if self.options.get_safe("with_openssl", False):
-            # self.options["libcurl"].with_openssl = True # deprecated in https://github.com/conan-io/conan-center-index/pull/2880
-            self.options["libcurl"].with_ssl = "openssl"
-        if self.options.get_safe("with_winssl", False):
-            # self.options["libcurl"].with_winssl = True # deprecated in https://github.com/conan-io/conan-center-index/pull/2880
-            self.options["libcurl"].with_ssl = "schannel"
+        # "auto" will be handled by cpr
+        if self._supports_ssl_library(ssl_library) and ssl_library in SSL_CURL_LIBS:
+            self.options["libcurl"].with_ssl = SSL_CURL_LIBS[ssl_library]
```

Comment:
I would instead raise in `validate`, if libcurl uses different SSL implementation because overriding user's desires is not something we should do.

Reply:
Okey doke. I did this primarily to mirror what was already there but I'll give this a fiddle and see how it works out.